packages/dds/migration-shim/src/shimDeltaConnection.ts

```diff
@@ -45,11 +51,61 @@ export class ShimDeltaConnection implements IDeltaConnection {
 		// preventing them from being processed by the tree delta handler, and the tree delta handler to process tree
 		// ops. Post migration v1 ops can be considered "shim" ops as they are dropped.
 		this.shimDeltaHandler.attachTreeDeltaHandler(handler);
-		if (!this.isShimDeltaHandlerAttachedToConnection) {
+		if (!this.shimDeltaHandler.attached) {
+			this.deltaConnection.attach(this.shimDeltaHandler);
+			this.shimDeltaHandler.markAttached();
+		}
+	}
+	public dirty(): void {
+		this.deltaConnection.dirty();
+	}
+
+	// This needs to be more thoroughly thought through. What happens when the source handle is changed?
+	public addedGCOutboundReference?(srcHandle: IFluidHandle, outboundHandle: IFluidHandle): void {
+		this.deltaConnection.addedGCOutboundReference?.(srcHandle, outboundHandle);
+	}
+}
+
+/**
+ * A delta connection that stamps ops with a particular channel attributes so that those ops won't get dropped
+ */
+export class StampDeltaConnection implements IDeltaConnection {
+	public constructor(
+		private readonly deltaConnection: IDeltaConnection,
+		private readonly shimDeltaHandler: IShimDeltaHandler,
+		private readonly attributes: IChannelAttributes,
+	) {}
+
+	public get connected(): boolean {
+		return this.deltaConnection.connected;
+	}
+
+	// This is for submitting v2 ops
+	public submit(messageContent: IStampedContents, localOpMetadata: unknown): void {
+		messageContent.fluidMigrationStamp = {
+			...this.attributes,
```

Comment:
Do we really need to stamp the whole attributes on here?  Aside from being larger than necessary, it's taking dependency on a structure/format that's not directly under the control of the shim.  Consider instead just stamping a number or boolean here.

Reply:
Probably a number is sufficient, but I would like there to be some variability in the stamping. Maybe I'm trying to optimize for a future that simply isn't important yet. If there's an upgrade to the shared tree, they're going to need a different stamp. It's quite likely SharedTree will try to migrate further, I'm trying to make sure we have some way of allowing them to figure out what has happened.